src/Deftype.hs

```diff
@@ -77,23 +77,45 @@ templatesForMembers _ _ _ _ _ = error "Shouldn't reach this case (invalid type d
 -- | Generate the templates for a single member in a deftype declaration.
 templatesForSingleMember :: TypeEnv -> Env -> [String] -> Ty -> (XObj, XObj) -> [((String, Binder), [XObj])]
 templatesForSingleMember typeEnv env insidePath p@(StructTy (ConcreteNameTy typeName) _) (nameXObj, typeXObj) =
-  let Just t = xobjToTy typeXObj
-      memberName = getName nameXObj
-  in [instanceBinderWithDeps (SymPath insidePath memberName) (FuncTy [RefTy p (VarTy "q")] (RefTy t (VarTy "q")) StaticLifetimeTy) (templateGetter (mangle memberName) t) ("gets the `" ++ memberName ++ "` property of a `" ++ typeName ++ "`.")
-     , if isTypeGeneric t
-       then (templateGenericSetter insidePath p t memberName, [])
-       else instanceBinderWithDeps (SymPath insidePath ("set-" ++ memberName)) (FuncTy [p, t] p StaticLifetimeTy) (templateSetter typeEnv env (mangle memberName) t) ("sets the `" ++ memberName ++ "` property of a `" ++ typeName ++ "`.")
-     , if isTypeGeneric t
-       then (templateGenericMutatingSetter insidePath p t memberName, [])
-       else instanceBinderWithDeps (SymPath insidePath ("set-" ++ memberName ++ "!")) (FuncTy [RefTy p (VarTy "q"), t] UnitTy StaticLifetimeTy) (templateMutatingSetter typeEnv env (mangle memberName) t) ("sets the `" ++ memberName ++ "` property of a `" ++ typeName ++ "` in place.")
-     ,instanceBinderWithDeps (SymPath insidePath ("update-" ++ memberName))
-                                                            (FuncTy [p, RefTy (FuncTy [t] t (VarTy "fq")) (VarTy "q")] p StaticLifetimeTy)
-                                                            (templateUpdater (mangle memberName))
-                                                            ("updates the `" ++ memberName ++ "` property of a `" ++ typeName ++ "` using a function `f`.")
-                                                            ]
+  case t of
+    -- Unit member types are special since we do not represent them in emitted c.
+    -- Instead, members of type Unit are executed for their side effects and silently omitted
+    -- from the produced C structs.
+    UnitTy ->
+      binders (FuncTy [RefTy p (VarTy "q")] UnitTy StaticLifetimeTy)
+              (FuncTy [p, t] p StaticLifetimeTy)
+              (FuncTy [RefTy p (VarTy "q"), t] UnitTy StaticLifetimeTy)
+              (FuncTy [p, RefTy (FuncTy [] UnitTy (VarTy "fq")) (VarTy "q")] p StaticLifetimeTy)
```

Comment:
Perhaps in the same vein it would be fine (or even desirable) to have to pass a function `(Fn [Unit] Unit)` here? For higher level code to work, I mean. In the concrete case it's clearly a nuissance...

Reply:
I run into a weird issue when I give the function a `Unit` argument:

```
out/main.c:6532:71: error: use of undeclared identifier 'Fn__LambdaEnv_void'
Foo Foo_update_MINUS_one(Foo p, Lambda *updater) { (*updater).env ? ((Fn__LambdaEnv_void)(*updater).callback)((*updater).env) : ((Fn___void)(*updater).callback)(); return p;}
```

Not sure where I need to update the code to ensure the env is declared